src/routes/(app)/documents/[id]-[slug].[format]/+page.ts

```diff
@@ -0,0 +1,43 @@
+// handle .html and other file extension routes from old versions of DocumentCloud
+
+import { error, redirect } from "@sveltejs/kit";
+import * as documents from "$lib/api/documents";
+
+export const trailingSlash = "never";
+
+export async function load({ params, fetch }) {
+  const { id, format } = params;
+
+  const { data: document, error: err } = await documents.get(id, fetch);
+
+  if (err) {
+    return error(err.status, err.message);
+  }
+
+  if (!document) {
+    return error(404, "Not found");
+  }
+
+  let url: URL;
+
+  switch (format) {
+    // for .html, redirect to embed
+    case "html":
+      url = documents.embedUrl(document);
+      return redirect(302, url);
+
+    case "txt":
+      url = documents.canonicalUrl(document);
+      url.searchParams.set("mode", "text");
+      return redirect(302, url);
+
+    // redirect to the PDF file, no-op for errors
+    case "pdf":
+      url = documents.pdfUrl(document);
+      return redirect(302, url);
+
+    // fallback: redirect to the canonical path, on the same host
```

Comment:
This is a great approach! My only design question is whether `.txt` should redirect to the text mode in an embed viewer instead of the full-site viewer.

Reply:
Historically, there wasn't a difference between embed and normal versions of the viewer (except in the very beginning), but the `.html` extension is baked into the WordPress plugin (by me, a long time ago, sorry) so it's still used in embeds.

This is the old documentation: https://web.archive.org/web/20180301171441/https://www.documentcloud.org/help/publishing#linking

All that said, I don't have much evidence of anyone using the `.txt.` or `.pdf` links, and it hasn't been supported in a few years, so I think we're ok like this, and we can adjust if needed.